src/language.rs

```diff
@@ -22,4 +51,40 @@ impl Language {
             )),
         }
     }
+
+    pub fn detect(filename: &str) -> FormatterResult<&str> {
+        // NOTE This extension search is influenced by Wilfred Hughes' Difftastic
+        // https://github.com/Wilfred/difftastic/blob/master/src/parse/guess_language.rs
+        if let Some(extension) = Path::new(filename).extension() {
```

Comment:
I think for the errors returned here a new variant `FormatterError::LanguageDetection` is warranted.

Reply:
Added in 6d9a1d4171c12e9aea